hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactNullablePrimitiveInteroperabilityTest.java

```diff
@@ -135,6 +137,64 @@ public void testWritePrimitiveReadNullable() {
         assertReadAsNullable(serializedRecord);
     }
 
+    @Test
+    public void testWritePrimitiveReadNullableCustomSerializer() {
+        FixedFieldsDTO fixedFieldsDTO = createFixedFieldsDTO();
+        CompactSerializationConfig compactSerializationConfig = new CompactSerializationConfig();
+        compactSerializationConfig.addSerializer(new FixedFieldsDTOSerializerReadingNullable());
+        SerializationService serializationService = new DefaultSerializationServiceBuilder()
+                .setSchemaService(schemaService)
+                .setConfig(new SerializationConfig().setCompactSerializationConfig(compactSerializationConfig))
+                .build();
```

Comment:
Maybe move this to a some method like `createSerializationService(Supplier<CompactSerializer> serializerSupplier)`?